Update custom.rst

manju-rn opened the pull request:
As advised in #847 the component must also be returned hence the documentation update

## Description:


**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

manju-rn committed:
Update custom.rst

As advised in #847 the component must also be returned hence the documentation update

OttoWinter commented on components/sensor/custom.rst:
```diff
@@ -333,23 +333,22 @@ Our YAML configuration needs an update too:
         auto my_sensor = new MyCustomSensor();
         App.register_component(my_sensor);
-        return {my_sensor->temperature_sensor, my_sensor->pressure_sensor};
+        return {my_sensor, my_sensor->temperature_sensor, my_sensor->pressure_sensor};
```
This is only true for classes that extend `Component` **and** `Sensor`.

The class above only inherits from `Component`, and thus should not appear in this list (it will lead to a compile error if you put it there).

manju-rn replied:
Agree. extended the Sensor to example MyCustomSensor

OttoWinter replied:
No, that is not good either.

While having a class both *extend* from Sensor and have Sensor members does work, it's not good code and confusing. The updated docs you pushed now will make Home Assistant add an empty entity with no name, when that is not used.

manju-rn replied:
I see your point.  How do you propose I resolve the situation what was mentioned in  #847 ?  I could not get a sensors to display in Home Assistant without passing the component.

OttoWinter replied:
@manju-rn In that case, yes you should add it to the list.

However, the problem is that the way your code works is *not* recommended. As explained in the previous comment, having a Sensor object also have Sensor members is not good API design.

manju-rn committed:
Update custom.rst

extended Sensor class for the example MyCustomSensor.  This would be logical since 1. Without this change, the user will not be able to see the sensors in HA. 2. The similar example class also has the Sensor extended 3. The whole document is on Custom Sensor !
